enclave_manager/avalon_enclave_manager/graphene/graphene_enclave_manager.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import os
+import sys
+import random
+
+from avalon_enclave_manager.base_enclave_manager import EnclaveManager
+from avalon_enclave_manager.work_order_processor_manager \
+    import WOProcessorManager
+from error_code.error_status import WorkOrderStatus
+from avalon_enclave_manager.graphene.graphene_enclave_info \
+    import SignupGraphene
+from avalon_enclave_manager.graphene.zmq_comm import ZmqCommunication
+
+logger = logging.getLogger(__name__)
+
+
+# -------------------------------------------------------------------------
+
+
+class GrapheneEnclaveManager(WOProcessorManager):
+    """
+    Manager class to handle Graphene based work order processing
+    """
+
+    def __init__(self, config):
+        """
+        Constructor for Graphene Enclave Manager
+
+        Parameters :
+            config: Configuration for Graphene Enclave Manager class
+        """
+        # zmq socket has to be created before calling super class constructor
+        # super class constructor calls _create_signup_data() which uses
+        # socket for communicating with Graphene worker.
+        graphene_zmq_url = config.get("EnclaveManager")["graphene_zmq_url"]
+        self.zmq_socket = ZmqCommunication(graphene_zmq_url)
+        super().__init__(config)
+        self.proof_data_type = config.get("WorkerConfig")["ProofDataType"]
+        self._identity = self._worker_id
+
+# -------------------------------------------------------------------------
+
+    def _manager_on_boot(self):
```

Comment:
Is boot time flow different than the enclave manager? otherwise the move it base class

Reply:
Base class is `WOProcessorManager`. And boot flow is different for WPE and Singleton. We could probably create a common function for Singleton and Graphene. But that would require changes to existing  files/functions which I wanted to avoid at this stage of release